src/neuroglancer/mesh/backend.ts

```diff
@@ -358,24 +365,33 @@ export class MeshSource extends ChunkSource {
     fragmentSource.meshSource = this;
   }
 
-  getChunk(objectId: Uint64) {
+  getChunk(objectId: Uint64, verifyFragments?: boolean|undefined) {
     const key = getObjectKey(objectId);
     let chunk = <ManifestChunk>this.chunks.get(key);
     if (chunk === undefined) {
+      if (verifyFragments === undefined) verifyFragments = true;
       chunk = this.getNewChunk_(ManifestChunk);
-      chunk.initializeManifestChunk(key, objectId);
+      chunk.initializeManifestChunk(key, objectId, verifyFragments);
       this.addChunk(chunk);
     }
     return chunk;
   }
 
-  getFragmentChunk(manifestChunk: ManifestChunk, fragmentId: FragmentId) {
-    let key = `${manifestChunk.key}/${fragmentId}`;
+  getFragmentChunk(manifestChunk: ManifestChunk, fragmentId: FragmentId, verifyFragment?: boolean|undefined) {
+    let key = fragmentId;
+    if (fragmentId.charAt(0) === '~') {
+      // extract segment ID from fragment ID
+      // use it as key, the rest is information for reading the fragment
+      let parts = fragmentId.substr(1).split(/:(.+)/);
```

Comment:
I don't believe the `(.+)` is necessary(?)
```suggestion
      let parts = fragmentId.substr(1).split(':');
```

Reply:
there can be multiple colons, need it to be split only on the first one
stole it from SO but let me know if there are other ways